Module-generation/scrape.py

```diff
@@ -0,0 +1,120 @@
+import os
+import json
+import requests
+from github import Github
+
+IndexDir="./meta-data/"
+os.mkdir(IndexDir)
+
+accessToken=os.environ.get('GIT_CREDS')
+g = Github(accessToken)
+
+
+user= g.get_user('Terasology')
+repo = user.get_repo("Index")
+
+contents = repo.get_contents("/distros/omega/gradle.properties")
+x=contents.decoded_content.decode()
+
+# Filter Modules and Convert it to array
+f = open('Modules.txt', 'w')
+f.write(x)
+f.close()
+f = open("Modules.txt", "r")
+lines=f.readlines()
+line=lines[1]
+word1='extraModules='
+word2='\n'
+wordreplace=line.replace(word1, "")
+wordreplace=wordreplace.replace(word2,"")
+Modules=wordreplace.split(",")
+
+
+# Collect Modules
+for i in range(0,len(Modules)):
+   if(i<len(Modules)):
+      repository= user.get_repo(Modules[i])
+      con=repository.get_contents("")
+      for content_file in con:
+          if(content_file.path=="module.txt"):
```

Comment:
I'd propose to directly try to get contents of `module.txt` here and properly handle the error that will be thrown if the file does not exist.

Reply:
This check is for the safer side, If modules.txt does not exist, just exit and it is not a module, I think directly getting content will make code look a bit complicated, If module.txt is not present it will return some error or null and that will require the try-catch block to handle it